src/app.ts

```diff
@@ -962,27 +961,26 @@ app.get(
     requireRole(GhostRole.Owner, GhostRole.Administrator),
     spanWrapper(createGetProfileFollowingHandler(accountService)),
 );
-app.get(
-    '/.ghost/activitypub/profile/:handle/posts',
-    requireRole(GhostRole.Owner, GhostRole.Administrator),
-    spanWrapper(createGetProfilePostsHandler(accountService, postRepository)),
-);
 app.get(
     '/.ghost/activitypub/thread/:post_ap_id',
     spanWrapper(createGetThreadHandler(postRepository, accountService)),
 );
 app.get(
     '/.ghost/activitypub/account/:handle',
     requireRole(GhostRole.Owner, GhostRole.Administrator),
-    spanWrapper(createGetAccountHandler(accountService, accountRepository)),
+    spanWrapper(
+        createGetAccountHandler(accountService, accountRepository, fedify),
+    ),
 );
 app.get(
-    '/.ghost/activitypub/posts',
+    '/.ghost/activitypub/posts/:handle',
```

Comment:
This endpoint returns a list of Activity for remote profiles and a list of Posts for local profiles. I went with this approach because mapping activity to postDTO here in backend and then converting it back to activity in the FE just felt like extra work.
I did try mapping it though, but it wasn’t super smooth and caused a bunch of UI issues that needed a lot of manual testing. Doing it this way made testing and shipping a lot easier.

Reply:
I know what you are saying, it does seem a little counter productive BUT we are eventually going to migrate the client to using `posts` and not activities so we are going to have to do this at some point. I think it would be worth while doing this now and not having to worry about the backend when we come to update the frontend. I think its also a bit of a smell that an endpoint could potentially return 2 different types of data and its not obvious why this is happening.

Could we not just use the same mapping fn thats used on the frontend?

Reply:
Added a mapping function [here](https://github.com/TryGhost/ActivityPub/pull/483/files#diff-eb3249a9a93286287064f8d43e8e503ae447978501b3af5b5f1c3d76eb1341a4R151).